authservice/auth_service.go

```diff
@@ -271,47 +302,43 @@ func NewTestAuthService(serviceID string, serviceHost string, dataLoader AuthDat
 	return auth, nil
 }
 
-// -------------------- DataLoader --------------------
+// -------------------- ServiceAccountLoader --------------------
 
-// AuthDataLoader declares an interface to load data from an auth service
-type AuthDataLoader interface {
+// ServiceAccountLoader declares an interface to load service account data from an auth service
+type ServiceAccountLoader interface {
 	// GetAccessToken gets an access token
 	GetAccessToken() error
 	// GetDeletedAccounts loads deleted account IDs
 	GetDeletedAccounts() ([]string, error)
-	ServiceRegLoader
 }
 
-//RemoteAuthDataLoaderImpl provides a AuthDataLoader implementation for a remote auth service
-type RemoteAuthDataLoaderImpl struct {
-	config RemoteAuthDataLoaderConfig
+//RemoteServiceAccountLoaderImpl provides a ServiceAccountLoader implementation for a remote auth service
+type RemoteServiceAccountLoaderImpl struct {
+	config RemoteServiceAccountLoaderConfig
 
 	accessToken AccessToken
 
 	timerDone               chan bool
 	getDeletedAccountsTimer *time.Timer
 
 	logger *logs.Logger
-
-	*RemoteServiceRegLoaderImpl
 }
 
-//RemoteAuthDataLoaderConfig represents a configuration for a remote data loader
-type RemoteAuthDataLoaderConfig struct {
+//RemoteServiceAccountLoaderConfig represents a configuration for a remote service account loader
+type RemoteServiceAccountLoaderConfig struct {
 	AuthServicesHost string // URL of auth services host
```

Comment:
We should avoid forcing the implementing service to pass the host in multiple interfaces by storing it in the auth service itself